### PR TITLE
Add parseSync

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

**Comment by hzoo** [comment]
If the tests (setup) are the same, maybe you can make it so you don't have to copy it for async/sync

**Comment by weikinhuang** [comment]
LGTM

**Comment by weikinhuang** [comment]
Though I agree with @hzoo about the test setup

File: test/parser/index.js [hunk 1]
```diff
@@ -2,29 +2,57 @@
...
+import { parse, parseSync } from '../../src/parser';
 import Focss from '../../src';
 
+function checkResult(styles, result) {
+  it('when parsed with parse', function(done) {
+    parse(styles)
+    .then(({ focssDescriptors }) => {
+      expect(focssDescriptors).toEqual(result);
+      done();
+    }).catch((e) => {
```
Comment (weikinhuang) [inline test/parser/index.js]: `.catch(done.fail)`

Comment (weikinhuang) [inline test/parser/index.js]: also next line

File: test/parser/index.js [hunk 2]
```diff
@@ -2,29 +2,57 @@
 
 import path from 'path';
 import fs from 'fs';
-import parse from '../../src/parser';
+import { parse, parseSync } from '../../src/parser';
 import Focss from '../../src';
 
+function checkResult(styles, result) {
+  it('when parsed with parse', function(done) {
```
Comment (weikinhuang) [inline test/parser/index.js]: actually, not a fan of putting `it` in here

File: test/parser/index.js [hunk 3]
```diff
@@ -82,204 +104,181 @@ describe('parser', function() {
...
-      done.fail(e);
+      const styles = fs.readFileSync(path.resolve(__dirname, 'fixtures', 'styles.focss'), { encoding: 'utf8' });
+      const result = '.foo{width:200px;}.foo:hover{max-width:800%;}nav .foo a.active, nav .foo a.__fake{margin-left:200px;}.foo 3{opacity:1;}.class1[data-id="3"]{float:none;margin-top:18px;width:18px;}.class1[data-id="4"]{float:left;margin-top:8px;width:8px;}.class2[data-id="1"]{max-width:400px;}.class2[data-id="2"]{max-width:600px;}@media screen and (max-width: 300px){.foo{width:200px;}.foo:hover{max-width:800%;}nav .foo a.active, nav .foo a.__fake{margin-left:200px;}.foo 3{opacity:1;}.class1[data-id="3"]{float:none;margin-top:18px;width:18px;}.class1[data-id="4"]{float:left;margin-top:8px;width:8px;}.class2[data-id="1"]{max-width:400px;}.class2[data-id="2"]{max-width:600px;}}';
+
+      it('when inserted descriptors are parsed with parse', function(done) {
+        parse(styles)
+        .then(({ focssDescriptors }) => {
+          processAndValidate(focssDescriptors, result);
+          done();
+        }).catch((e) => {
```
Comment (weikinhuang) [inline test/parser/index.js]: here too

File: test/parser/index.js (hunk 2 above)
Comment (kaicataldo) [inline test/parser/index.js]: What's the downside?

File: test/parser/index.js [hunk 4]
```diff
@@ -2,29 +2,56 @@
...
-      expect(focssDescriptors.length).toEqual(2);
-      done();
-    }).catch((e) => {
-      done.fail(e);
+    it('when parsed with parse', function(done) {
+      parse(styles)
+      .then(({ focssDescriptors }) => {
+        checkDescriptors(focssDescriptors);
+        done();
+      }).catch((e) => {
```
Comment (weikinhuang) [inline test/parser/index.js]: and here

Comment (weikinhuang) [inline test/parser/index.js]: actually, you can do what i did in `be-hippo`: `.then(done, done.fail)`

File: test/parser/index.js (hunk 2 above)
Comment (weikinhuang) [inline test/parser/index.js]: the describe block looks super weird, as if there's no tests in there

**Comment by hzoo** [comment]
👍

**Comment by weikinhuang** [comment]
👍

**Pull request merged**